core/src/main/java/org/apache/druid/math/expr/Expr.java

```diff
@@ -304,7 +334,37 @@ public BindingDetails withArrayArguments(Set<Expr> arrayArguments)
       return new BindingDetails(
           ImmutableSet.copyOf(Sets.union(freeVariables, arrayIdentifiers)),
           scalarVariables,
-          ImmutableSet.copyOf(Sets.union(arrayVariables, arrayIdentifiers))
+          ImmutableSet.copyOf(Sets.union(arrayVariables, arrayIdentifiers)),
+          hasInputArrays || arrayArguments.size() > 0,
```

Comment:
Is it possible for `arrayArguments.size()` to be greater than zero, but `hasInputArrays` to be false? If so how? (Related to previous comment, I think)

Reply:
I don't think so, this can probably just check `hasInputArrays`, but I will confirm by removing the size check and seeing what happens to the tests

Reply:
Oh i wasn't paying attention to which function this was, it is possible in this specific context and it's unrelated to the other thing. 

Since this is a sort of 'builder' method it is hypothetically possible for this method to be called when adding the initial set of array columns on a binding details, without explicitly calling `withArrayInputs`. This `||` then ensures that if even if you forgot to call `withArrayInputs` the binding details being constructed are correct, becase the `hasInputArrays` in this case is the value of the existing binding before the array arguments are added.